### ISSUE TITLE
Proposal: Allow messages after directives

### ISSUE THREAD

**Issue by Kreyren** [koalaman/shellcheck#1771]
#### For bugs
- Rule ID: **SC1072**
- My shellcheck version: **0.7.0**
- [X] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [X] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [X] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh 
debug() {
	message="$1"

	# shellcheck disable=SC2154 - Variable 'debug' is set by the end-user
	[ -n "$debug" ] && printf 'DEBUG: %s\n' "$message"

	unset message
}

```

#### Here's what shellcheck currently says:

`Expected '=' after directive key. Fix any mentioned problems and try again. [SC1072]`

#### Here's what I wanted or expected to see:

Expecting highlighted message to be valid for shellcheck
```
# shellcheck disable=SC2154 - Variable 'debug' is set by the end-user
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Reasoning:** Less multi-line comments with reasoning to use `disable` directive compared to:
```sh
#!/bin/sh
debug() {
	message="$1"

	# Variable 'debug' is set by the end-user
	# shellcheck disable=SC2154
	[ -n "$debug" ] && printf 'DEBUG: %s\n' "$message"

	unset message
}
```

Which i believe makes the code less readable especially when there is more comments above the code.

**Comment by matthewpersico** [comment]
```
#!/bin/sh 
debug() {
	message="$1"

	# shellcheck disable=SC2154 # Variable 'debug' is set by the end-user
	[ -n "$debug" ] && printf 'DEBUG: %s\n' "$message"

	unset message
}

$ shellcheck myscript
No issues detected!
```
Use another comment character. That is what I do.

**Comment by Kreyren** [comment]
Resolved by @matthewpersico's solution, thanks!

**Comment by Kreyren** [comment]
Apparently this no longer works?

See https://github.com/gitpod-io/dockerfreeze/pull/54/checks?check_run_id=485963696

```sh
In tools/dockerfreeze line 6:
    # shellcheck disable=SC2034 # FIXME: Are cmds used?
                                ^-- SC1107: This directive is unknown. It will be ignored.
                                  ^-- SC1107: This directive is unknown. It will be ignored.
                                         ^-- SC1107: This directive is unknown. It will be ignored.
                                             ^-- SC1107: This directive is unknown. It will be ignored.
                                                  ^-- SC1107: This directive is unknown. It will be ignored.
```

**Comment by Kreyren** [comment]
Resolved: Github is using version: 0.4.6  which apparently does not support this
